__tests__/events-api.ts

```diff
@@ -0,0 +1,114 @@
+import { useTask, addEventHandler, removeEventHandler } from "../src/index";
+import { EventTargetOptions } from "../src/types/index";
+import { mockSetup } from "../test-utils/components";
+
+describe("public events api", () => {
+  test("using events api outside of a component", async () => {
+    let fired = false;
+    const handler = () => {
+      fired = true;
+    };
+
+    const eventHandler = {
+      target: EventTargetOptions.OnError,
+      key: "test",
+      handler,
+    };
+
+    addEventHandler(eventHandler);
```

Comment:
Thanks for the changes 👍 

Here's how the usage could be further improved (in my opinion) to reduce some verbosity. 

```ts
addTaskEventHandler('error', () => {
 //...
});
```

No `key`. Instead the function itself is used. The more it is consistent with `addEventListener` the better. People already know that pattern.

```ts
const handlers = {
  'error': [...] // array of functions,
  'whatever': [...]
};
```

Or do you see some downsides to this approach?